### PR TITLE
[pkg/stanza] Switch from SugaredLogger to Logger

### PR DESCRIPTION
**Pull request by djaglowski**
#32662 updated many exported functions to accept `component.TelemetrySettings` instead of `zap.SugaredLogger`. This PR continues from there by passing `component.TelemetrySettings` deeper into the inner packages, and migrates from using `zap.SugaredLogger` to instead using `zap.Logger` (as provided by `component.TelemetrySettings`).

### CODE REVIEW

**Comment by djaglowski** [comment]
This is just waiting on a review @open-telemetry/collector-contrib-approvers @ChrsMark

File: .chloggen/pkg-stanza-rm-sugared.yaml
Comment (mx-psi) [inline .chloggen/pkg-stanza-rm-sugared.yaml:4]: Is changing the logging format a breaking change?

File: pkg/stanza/fileconsumer/internal/header/reader.go
Comment (mx-psi) [inline pkg/stanza/fileconsumer/internal/header/reader.go:22]: Not necessary, but for consistency with other structs, why not store the `component.TelemetrySettings` here instead?

File: pkg/stanza/fileconsumer/internal/reader/factory.go
Comment (mx-psi) [inline pkg/stanza/fileconsumer/internal/reader/factory.go:32]: Do we need this to be a public field?

File: pkg/stanza/operator/helper/operator.go
Comment (mx-psi) [inline pkg/stanza/operator/helper/operator.go:78]: Same as above, why not store `component.TelemetrySettings` here?

File: pkg/stanza/pipeline/directed.go
Comment (mx-psi) [inline pkg/stanza/pipeline/directed.go:30]: We are not super consistent with deprecation of this vs outright removal. Is there any reason for it?

File: pkg/stanza/testutil/util.go
Comment (mx-psi) [inline pkg/stanza/testutil/util.go:19]:
```suggestion
// Deprecated: [v0.100.0] Use zaptest.NewLogger directly instead
```

File: .chloggen/pkg-stanza-rm-sugared-api.yaml
Comment (ChrsMark) [inline .chloggen/pkg-stanza-rm-sugared-api.yaml:4]: Mostly for my understanding: isn't this also a breaking change since the `Operator` interface changes and specifically the `Logger() *zap.SugaredLogger` -> `Logger() *zap.Logger`?

File: pkg/stanza/pipeline/directed.go
Comment (ChrsMark) [inline pkg/stanza/pipeline/directed.go:29]: Same here, I think it should target the next release (`v0.101.0`)?

File: .chloggen/pkg-stanza-rm-sugared.yaml
Comment (djaglowski) [reply .chloggen/pkg-stanza-rm-sugared.yaml:4]: Potentially, for users who are relying on semantic structure of logs. It's pretty minor but I figured in this case maybe it was worth noting because that change affect many log statements.

File: .chloggen/pkg-stanza-rm-sugared-api.yaml
Comment (djaglowski) [reply .chloggen/pkg-stanza-rm-sugared-api.yaml:4]: Yeah, good point.

File: pkg/stanza/fileconsumer/file.go
Comment (djaglowski) [reply pkg/stanza/fileconsumer/file.go:25]: Yup, PR just didn't get reviewed before the release. Updated.

File: pkg/stanza/fileconsumer/internal/header/reader.go
Comment (djaglowski) [reply pkg/stanza/fileconsumer/internal/header/reader.go:22]: Yeah, good point. Easy enough to do anyways

File: pkg/stanza/fileconsumer/internal/reader/factory.go
Comment (djaglowski) [reply pkg/stanza/fileconsumer/internal/reader/factory.go:32]: Unfortunately yes, for now at least, because we create the factory manually. (Rather than having a builder or factory for the factory). However, it is in an very internal package of `fileconsumer` so it's not actually increasing our surface area.

File: pkg/stanza/operator/helper/operator.go
Comment (djaglowski) [reply pkg/stanza/operator/helper/operator.go:78]: Done

File: pkg/stanza/pipeline/directed.go
Comment (djaglowski) [reply pkg/stanza/pipeline/directed.go:30]: I tried to make an effort to minimize breaking changes but I think in a few cases leaving the logger in place could be less clear since it would just be unused. Are there are any instances in particular you think I should revisit?

File: pkg/stanza/fileconsumer/internal/reader/factory.go
Comment (mx-psi) [reply pkg/stanza/fileconsumer/internal/reader/factory.go:32]: Makes sense, thanks for clarifying

**Review by mx-psi — APPROVED** [review]

File: pkg/stanza/pipeline/directed.go
Comment (mx-psi) [reply pkg/stanza/pipeline/directed.go]: I think I misread this, because I can't find said instances now :smile: Marking as resolved

**Comment by fordneild** [comment]
This seems to have broken drop_quiet and send_quiet from what I can tell. Now my failed "move" operator is logging even though i have it on send_quiet

**Comment by mx-psi** [comment]
> This seems to have broken drop_quiet and send_quiet from what I can tell. Now my failed "move" operator is logging even though i have it on send_quiet

@fordneild would you mind opening a separate issue for this? 

cc @djaglowski

**Pull request merged**
